### PR TITLE
34 profile page react component

### PR DESCRIPTION
**Pull request by oshtohryn**
## TRELLO

* [Main Trello ticket](https://trello.com/c/52N7Ukep/34-profile-page-react-component)

## Code reviewers

- [ ] @black-art
- [ ] @hrytskivr
- [x] @KarnaushenkoVO
- [ ] @PanLeon
- [ ] @spycheast
- [ ] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

JavaScript

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/static/src/profile/profileEdit.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+export class TextBlock extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = props.data;
+    };
+
+    getProfile = () => {
+    return axios.get('/api/v1/profile/')
+    .then(response => this.setState({state: response.data}))
```
Comment (KarnaushenkoVO) [inline myTrip/static/src/profile/profileEdit.js]: t o service

File: myTrip/static/src/profile/profileEdit.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    componentDidMount(){
+        this.getProfile()
+    }
+
+    onChange = (event, newValue) => {
+        this.setState({[event.target.name]: newValue});
+    };
+
+    putProfile = (name, surname, age, gender, hobbies) => {
+        return axios.put('/api/v1/profile/', {
```
Comment (KarnaushenkoVO) [inline myTrip/static/src/profile/profileEdit.js]: to service

File: myTrip/static/src/profile/profileEdit.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          floatingLabelStyle={styles.LabelStyle}
+          floatingLabelFocusStyle={styles.LabelStyle}
+        />
+        <FlatButton onTouchTap={this.profileEdit} 
+        label="Edit profile" primary={true} fullWidth={true} 
+        rippleColor={blue500} />
+      </div>
+    );
+};
+}
```
Comment (KarnaushenkoVO) [inline myTrip/static/src/profile/profileEdit.js]: add newline

File: myTrip/static/src/profile/profileButtons.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import RaisedButton from 'material-ui/RaisedButton';
+import './profile.less';
+import { styles } from './profile.style.js'
+
+
+export class Buttons extends React.Component {
+  render() {
+    return(
+      <div className='divbutton'>
```
Comment (lhalam) [inline myTrip/static/src/profile/profileButtons.js]: map

File: myTrip/static/src/profile/profile.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+
+import './profile.less';
+
+
+export default class Profile extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+        email: 'joegrecarlin@gmail.com',
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.js]: remove the magic numbers and strings

File: myTrip/static/src/profile/profile.service.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios"
+
+export let getProfile = () => {
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.service.js]: Why `let` and not `const`?

File: myTrip/static/src/profile/profile.service.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    return axios.get('/api/v1/profile/')
+    .then(response => this.setState({state: response.data}))
+    .catch(error => console.log(error))
+    }
+
+export let putProfile = (name, surname, age, gender, hobbies) => {
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.service.js]: Why `let` and not `const`?

File: myTrip/static/src/utils.js [hunk 8]
```diff
@@ -15,3 +17,21 @@ export function emailIsNotValid(email) {
...
+
+}
+
+export function onlyDigit(numbers) {
+    if (DIGIT_REGEXP.test(numbers) == false) {
+        return false;
+    } else {
+        return true
+    }
+}
```
Comment (lhalam) [inline myTrip/static/src/utils.js]: end line

**Review by lhalam — COMMENTED** [review]
To Do

**Pull request closed without merging**